File: frontend/src/project/project.ts
import IAccessTokenData from "../types/access.token";
import { IAppData } from "../types/app";

export default interface IProject {
    id: number,
    identifier: string,
    name: string,
    companyId: number,
    accessTokens: IAccessTokenData[],
    apps: IAppData[],
}

File: storage/models/project.go
package models

type Project struct {
	Model
	Identifier   string         `json:"identifier"`
	Name         string         `json:"name"`
	CompanyID    uint           `json:"companyId"`
	Company      *Company       `json:"-"`
	AccessTokens []*AccessToken `json:"accessTokens"`
	Users        []*User        `json:"users" gorm:"many2many:user_projects;"`
	Apps         []*App         `json:"apps"`
}
